workflow/src/main/java/bio/terra/tanagra/workflow/TextSearchUtils.java

```diff
@@ -0,0 +1,90 @@
+package bio.terra.tanagra.workflow;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.beam.sdk.transforms.Distinct;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.join.CoGbkResult;
+import org.apache.beam.sdk.transforms.join.CoGroupByKey;
+import org.apache.beam.sdk.transforms.join.KeyedPCollectionTuple;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.TypeDescriptors;
+
+public final class TextSearchUtils {
+  private TextSearchUtils() {}
+
+  /**
+   * Concatenate all search strings for a node into a single string. e.g. (123, "apple pie"), (123,
+   * "fruit dessert"), (123, "apple crumble") => (123, "apple pie , fruit dessert , apple crumble")
+   *
+   * @param allNodes a collection of all (node) entity instances
+   * @param searchStrings a collection of all (node, text) mappings, where there may be zero or
+   *     multiple elements per node
+   * @return a collection of (node, text) mappings, where there is exactly one element per node
+   */
+  public static PCollection<KV<Long, String>> buildSingleSearchString(
```

Comment:
nit:  would `concatenateSearchStrings` or  `concatenateSearchStringsByKey` be more accurate?

Reply:
Yes, that's clearer thanks. Changed to `concatenateSearchStringsByKey`.